src/timeoffset.rs

```diff
@@ -0,0 +1,362 @@
+use crate::{Epoch, TimeScale, Unit};
+
+/// Errors during [TimeOffset] exploitation
+#[derive(Debug)]
+pub enum TimeOffsetError {
+    /// Timescales should not be identical when defining a new [TimeOffset]!
+    IdenticalTimescales,
+    /// Epoch is not expressed in one of the two supported [TimeScale]s!
+    NotSupportedTimescale,
+    /// Epoch should be expressed in the left-hand side [TimeScale].
+    InvalidTimescale,
+    /// [TimeOffset] is outdated and should not apply (difference is too large): should be updated weekly at least!
+    OutdatedTimeOffset,
+}
+
+/// [TimeOffset] used in [TimeShift]ing operations
+#[derive(Copy, Clone, PartialEq)]
+pub struct TimeOffset {
+    /// Left hand side (compared) [TimeScale]
+    lhs: TimeScale,
+    /// Right hand side (reference) [TimeScale]
+    rhs: TimeScale,
+    /// Weekly reference time (counter, nanoseconds)
+    t_ref_nanos: (u32, u64),
+    /// Polynomials (s, s.s⁻¹, s.s⁻²)
+    polynomials: (f64, f64, f64),
+}
+
+impl TimeOffset {
+    /// Define a new [TimeOffset] from this reference [Epoch] expressed
+    /// in left-hand side [TimeScale] to rhs [TimeScale]
+    pub fn from_reference_epoch(
+        t_ref: Epoch,
```

Comment:
```suggestion
        ref_epoch: Epoch,
```
That's the nomenclature used in a few parts of the code.